include/envoy/server/listener_manager.h

```diff
@@ -102,6 +102,16 @@ class ListenerComponentFactory {
  */
 class ListenerManager {
 public:
+  // Indicates listeners to stop.
+  enum class StopListenersType {
+    // Listeners in the inbound direction are only stopped.
+    InboundOnly,
+    // All listeners are stopped.
+    All,
+    // Listeners are not stopped by direction.
```

Comment:
It's a little odd to me that we have a stop type of "none" when passed as an API parameter. Can we just remove this and use an optional for the internal state if nothing is stopped yet?

Reply:
I have added it earlier to support other criteria and that time direction can be None. But now it is not needed. removed